### DISCUSSION TITLE
Solution for unable to get local issuer certificate for using Github Copilot in Visual Studio 2022

### DISCUSSION THREAD

**Discussion by dukewillbe185** [community/community#111175] [Copilot Conversations]
### Select Topic Area

General

### Body

If anyone has encountered this:

Have got the liscense to access Github Copilot in Visual Studio 2022, installed it, but got the issue: unable to get local issuer certificate

The point is: you need Zscaler Root CA.perm

[](https://stackoverflow.com/questions/77795127/copilot-behind-zscaler-error-on-ghost-text-request-fetcherror-unable-to-get/78118959#78118959)

<img width="742" alt="image" src="https://github.com/community/community/assets/81335199/10c63fd7-3b31-4d56-97e5-d7f4c01c7e00">

The answer from above link provides us a good way to solve this problem, but the point is you might cannot access the related link:
https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex

Error would be like this:

<img width="1107" alt="image" src="https://github.com/community/community/assets/81335199/d604a0f1-99d0-4610-8567-b9916b1bf78a">

Apparently this link has been expired, and Github has published this news in January from Github Blog:
<img width="1021" alt="image" src="https://github.com/community/community/assets/81335199/be61f61f-420b-4d98-97ae-7048843bde13">

So it seems like we're in a puzzle, but does that mean there's no way to solve it? Of course not

The key part for solving this problem is the file: Zscaler Root CA.pem, the expired link just showed us a way to download it, but we can actually get it in our computer, methods are below:

<img width="782" alt="image" src="https://github.com/community/community/assets/81335199/dbefa68f-4fef-4bc9-9729-6441c74d420d">

Next is becoming easier, find the related certi in your laptop and export it to your local environment, then follow the steps from above link(StackOverFlow) and set up everything.

After done, restart Visual Studio 2022 and everything works.

**Comment by wns-Pawan** [comment]
It worked for me. Thanks for the solution.
